src/libpmemfile-posix/inode.c

```diff
@@ -232,7 +232,7 @@ vinode_unref(PMEMfilepool *pfp, struct pmemfile_vinode *vinode)
 			parent = vinode->parent;
 		}
 
-		if (vinode != pfp->root)
+		if (vinode->parent != vinode)
```

Comment:
!vinode_is_root ?

Reply:
As it is now, the `vinode_is_root` function first checks if it is a directory (if it is not a dir, it can't be root), and for that, it uses vinode->inode->flags or whatnot. Which is a problem if vinode->inode is free'd already -- valgrind complains.